lib/bundlex/doxygen.ex

```diff
@@ -0,0 +1,166 @@
+defmodule Bundlex.DoxygenError do
+  defexception [:message]
+end
+
+defmodule Bundlex.Doxygen do
+  @moduledoc """
+  Module responsible for generating doxygen documentation for Bundlex projects.
+  """
+
+  alias Bundlex.Project
+
+  @type doxygen_t :: %{
+          project_name: String.t(),
+          doxyfile_path: String.t(),
+          doxygen_path: String.t(),
+          page_path: String.t()
+        }
+
+  @doc """
+  Prepares struct with all necessary filepaths for the native documentation
+  """
+  @spec doxygen(Project.t()) :: doxygen_t()
+  def doxygen(project) do
+    project_name = Atom.to_string(project.app)
+
+    %{
+      project_name: project_name,
+      doxyfile_path: doxyfile_path(project),
+      doxygen_path: doxygen_path(project),
+      page_path: page_path(project)
+    }
+  end
+
+  defp doxyfile_path(project) do
+    project_name = "#{project.app}"
+    project_dirpath = Path.join(["doc", "doxygen", project_name])
+    Path.join(project_dirpath, "Doxyfile")
+  end
+
+  defp doxygen_path(project) do
+    project_name = "#{project.app}"
+    Path.join(["doc", "doxygen", project_name])
+  end
+
+  defp page_path(project) do
+    Path.join(["pages", "doxygen", "#{project.app}.md"])
+  end
+
+  @doc """
+  Generates doxyfile in the c_src/project directory for Bundlex project.
+  """
+  @spec generate_doxyfile(doxygen_t()) :: :ok
+  def generate_doxyfile(doxygen) do
+    create_doxyfile_template(doxygen)
+
+    keywords_to_change = keywords_to_change(doxygen)
+    update_doxyfile_keywords(doxygen, keywords_to_change)
+  end
+
+  defp create_doxyfile_template(doxygen) do
+    ensure_doxygen_dir_existence(doxygen)
+    cmd_bundlex(["-g", doxygen.doxyfile_path])
+  end
+
+  defp cmd_bundlex(args) do
+    with {_res, 0} <-
+           System.cmd("doxygen", args) do
+      :ok
+    else
+      {output, status} ->
+        "Running doxygen command with args: #{inspect(args)} failed with exit code: #{inspect(status)} and output: #{inspect(output)}"
+    end
+  rescue
+    e in RuntimeError ->
+      reraise e, __STACKTRACE__
+
```

Comment:
This match on `RuntimeError` is still there, I would delete it at all, it does not improve anything, only might make error less readable, because of type override

Reply:
Done